components/emoji_picker/utils/index.ts

```diff
@@ -0,0 +1,342 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import isEmpty from 'lodash/isEmpty';
+
+import {Emoji, EmojiCategory} from 'mattermost-redux/types/emojis';
+
+import {
+    Categories,
+    Category,
+    CategoryOrEmojiRow,
+} from 'components/emoji_picker/types';
+
+import * as EmojiUtils from 'utils/emoji.jsx';
+import {getSkin} from 'utils/emoticons';
+import {compareEmojis} from 'utils/emoji_utils';
+import EmojiMap from 'utils/emoji_map';
+
+import {
+    EMOJI_PER_ROW,
+    CATEGORY_HEADER_ROW,
+    EMOJIS_ROW,
+    SEARCH_RESULTS,
+    CUSTOM,
+    RECENT,
+} from 'components/emoji_picker/constants';
+
+function sortEmojis(
+    emojis: Emoji[],
+    recentEmojiStrings: string[],
+    filter: string,
+) {
+    const recentEmojis: Emoji[] = [];
+    const emojisMinusRecent: Emoji[] = [];
+
+    Object.values(emojis).forEach((emoji) => {
+        let emojiArray = emojisMinusRecent;
+
+        const alias = emoji.short_names ? emoji.short_names : [emoji.name];
+        for (let i = 0; i < alias.length; i++) {
+            if (recentEmojiStrings.includes(alias[i].toLowerCase())) {
+                emojiArray = recentEmojis;
+            }
+        }
+
+        emojiArray.push(emoji);
+    });
+
+    return [
+        ...recentEmojis.sort((firstEmoji, secondEmoji) =>
+            compareEmojis(firstEmoji, secondEmoji, filter),
+        ),
+        ...emojisMinusRecent.sort((firstEmoji, secondEmoji) =>
+            compareEmojis(firstEmoji, secondEmoji, filter),
+        ),
+    ];
+}
+
+function getFilteredEmojis(
+    allEmojis: Record<string, Emoji>,
+    filter: string,
+    recentEmojisString: string[],
+): Emoji[] {
+    const emojis = Object.values(allEmojis).filter((emoji) => {
+        const alias = emoji.short_names ? emoji.short_names : [emoji.name];
+
+        for (let i = 0; i < alias.length; i++) {
+            if (alias[i].toLowerCase().includes(filter)) {
+                return true;
+            }
+        }
+
+        return false;
+    });
+
+    return sortEmojis(emojis, recentEmojisString, filter);
+}
+
+function convertEmojiToUserSkinTone(
+    emoji: Emoji,
+    emojiSkin: string,
+    userSkinTone: string,
+): Emoji {
+    if (emojiSkin === userSkinTone) {
+        return emoji;
+    }
+
+    let newEmojiId = '';
+
+    // If its a default (yellow) emoji, get the skin variation from its property
+    if (emojiSkin === 'default') {
+        newEmojiId = emoji?.skin_variations?.[userSkinTone]?.unified ?? '';
+    } else if (userSkinTone === 'default') {
+        newEmojiId = emoji?.unified?.replaceAll(`-${emojiSkin}`, '') ?? '';
```

Comment:
Make sure to copy your other changes over here once those are fixed on master

Reply:
✅  Yes copied